src/util-streaming-buffer.c

```diff
@@ -0,0 +1,689 @@
+/* Copyright (C) 2015-2016 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "util-streaming-buffer.h"
+#include "util-unittest.h"
+#include "util-print.h"
+
+/**
+ * \file
+ *
+ * \author Victor Julien <victor@inliniac.net>
+ *
+ *  \brief Streaming Buffer API
+ */
+
```

Comment:
Ack on the docs.

Wrt other use. It's mostly meant for cases where the data we're tracking is (potentially) too large to buffer completely. We're sliding through it. This we already do in HTTP, FILE and TCP currently, but with lots of fragmentation and reassembly overhead. This tries to streamline that.